OpenRA.Mods.Common/Traits/Render/WithAttackAnimation.cs

```diff
@@ -53,12 +54,20 @@ public WithAttackAnimation(ActorInitializer init, WithAttackAnimationInfo info)
 			armament = init.Self.TraitsImplementing<Armament>()
 				.Single(a => a.Info.Name == info.Armament);
 			wsb = init.Self.Trait<WithSpriteBody>();
+
+			if (init.Self.TraitOrDefault<AttackTurreted>() != null)
```

Comment:
We're looking for `AttackTurreted` in this line and for `WithSpriteTurret` in the line below. However, it should be possible to use `TraitOrDefault<WithSpriteTurret>` without looking for `AttackTurreted` in the first place. As you're null-checking `wst` everywhere anyway, you should be able to drop the `if` entirely as well. (I.e. make it just `wst = init.Self.TraitOrDefault<WithSpriteTurret>();`.)